python/glow/wgr/tests/test_pivot_for_gwas.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pyspark.sql import Row
+from pyspark.sql.functions import *
+from pyspark.sql.utils import AnalysisException
+import glow
+from glow.wgr import functions
+
+
+def test_error_too_many_levels(spark):
+    df = pd.DataFrame(columns=['c1', 'c2', 'c3'])
+    df = df.set_index(['c1', 'c2', 'c3'])
+    with pytest.raises(ValueError):
+        functions.pivot_for_gwas(spark, df)
+
+
+def test_pivot_loco(spark):
+    df = pd.read_csv('test-data/wgr/ridge-regression/level1YHatLoco.csv')
+    df = df.set_index(['sample_id', 'contigName'])
```

Comment:
Instead of re-setting `df` here, can you `set_index` immediately following the `read_csv` or use `set_index(inplace=True)`? Also, we likely expect the sample IDs and contig names to be strings, which can be done as follows:
```
level1_yhat_loco_df = pd.read_csv(f'{data_root}/level1YHatLoco.csv',
                                  dtype={
                                      'sample_id': 'str',
                                      'contigName': 'str'
                                  }).set_index(['sample_id', 'contigName'])
```
We may want to add a check for this in the pivoting function.

Reply:
I actually don't think that we need to or should require this. If someone wants to treat contig names as ints, for instance, that should be totally fine.